lib/nerves_hub_link/client.ex

```diff
@@ -174,6 +182,19 @@ defmodule NervesHubLink.Client do
     _ = apply_wrap(mod(), :handle_error, [data])
   end
 
+  @doc """
+  This function is called internally by NervesHubLink to notify clients of disconnects.
+  """
+  def reconnect_backoff() do
+    backoff = mod().reconnect_backoff()
+
+    if is_nil(backoff) do
+      NervesHubLink.Backoff.delay_list(1000, 60000, 0.50)
+    else
+      backoff
+    end
```

Comment:
I switched to success logic, and I think that I'm for using `apply_wrap/3`. The reason is that if anything goes wrong with getting the `backoff` values, I think that crashing causes an immediate restart and retry. That seems undesirable. I also think that `apply_wrap/3` really should log an error when it catches something (future PR, though, since that not your problem).

```suggestion
    if is_list(backoff) do
      backoff
    else
      NervesHubLink.Backoff.delay_list(1000, 60000, 0.50)
    end
```